Content.Server/Drone/DroneSystem.cs

```diff
@@ -0,0 +1,145 @@
+            SubscribeLocalEvent<DroneComponent, DisarmAttemptEvent>(OnDisarmAttempt);
+            SubscribeLocalEvent<DroneComponent, DropAttemptEvent>(OnDropAttempt);
+            SubscribeLocalEvent<DroneComponent, IsUnequippingAttemptEvent>(OnUnequipAttempt);
+            SubscribeLocalEvent<DroneComponent, ExaminedEvent>(OnExamined);
+            SubscribeLocalEvent<DroneComponent, MindAddedMessage>(OnMindAdded);
+            SubscribeLocalEvent<DroneComponent, MindRemovedMessage>(OnMindRemoved);
+        }
+
+        private void OnExamined(EntityUid uid, DroneComponent component, ExaminedEvent args)
+        {
+            if (args.IsInDetailsRange)
+            {
+                if (EntityManager.TryGetComponent<MindComponent>(uid, out var mind) && mind.HasMind)
+                {
+                    args.PushMarkup(Loc.GetString("drone-active"));
+                }
+                else
+                {
+                    args.PushMarkup(Loc.GetString("drone-dormant"));
+                }
+            }
+        }
+
+        private void OnMobStateChanged(EntityUid uid, DroneComponent drone, MobStateChangedEvent args)
+        {
+            if (args.Component.IsDead())
+            {
+                foreach (var item in drone.ToolUids)
+                {
+                        EntityManager.DeleteEntity(item);
+                }
+                _explosions.SpawnExplosion(uid);
+                EntityManager.DeleteEntity(uid);
+            }
+        }
+
+        private void OnDisarmAttempt(EntityUid uid, DroneComponent drone, DisarmAttemptEvent args)
+        {
+            EntityManager.TryGetComponent<HandsComponent>(args.TargetUid, out var hands);
+            var item = hands.GetActiveHandItem;
+            if (EntityManager.TryGetComponent<DroneToolComponent>(item?.Owner, out var itemInHand))
+            {
+            args.Cancel();
+            }
+        }
+
+        private void OnMindAdded(EntityUid uid, DroneComponent drone, MindAddedMessage args)
+        {
+            EntityManager.TryGetComponent<TagComponent>(uid, out var tagComp);
+            UpdateDroneAppearance(uid, DroneStatus.On);
+            tagComp.AddTag("DoorBumpOpener");
+            _popupSystem.PopupEntity(Loc.GetString("drone-activated"), uid, Filter.Pvs(uid));
+
+            if (drone.alreadyAwoken == false)
+            {
+                var spawnCoord = Transform(uid).Coordinates;
+
+                if (drone.Tools.Count == 0) return;
+
+                if (EntityManager.TryGetComponent<HandsComponent>(uid, out var hands))
+                {
+                   foreach (var entry in drone.Tools)
+                    {
+                        var item = EntityManager.SpawnEntity(entry.PrototypeId, spawnCoord);
+                        EntityManager.AddComponent<DroneToolComponent>(item);
+                        hands.PutInHand(item);
+                        drone.ToolUids.Add(item);
+                    }
+                }
```

Comment:
you should probably do some check that the number of hands is equal to/greater than the number of tools being spawned